.github/workflows/testing-alpine.yml

```diff
@@ -0,0 +1,35 @@
+name: build and test on AlpineLinux
+
+on:
+  push:
+    branches: [ master, devel ]
+  pull_request:
+    branches: [ master, devel ]
+
+jobs:
+  testing:
+    strategy:
+      fail-fast: false
+      matrix:
+        tag: ['3.20','3.19','3.18','3.17']
+        build-type: [Debug,Release]
+
+    runs-on: ubuntu-latest
+
+    container: alpine:${{ matrix.tag }}
+
+    steps:
+      - run: apk add git bash libc++-dev g++ cmake ninja
+
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - run: cmake -S . -B build.d -G Ninja -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_COLOR_MAKEFILE=ON
```

Comment:
Any reason to use ninja instead of make?

Reply:
both ninja and make are ok.

Reply:
Ok, testing with various generators shouldn't hurt.